File: files/CH-135.py
import pandas as pd
import re

path = "CH-135 Identify the Pattern.xlsx"
input = pd.read_excel(path, usecols="B:D", skiprows=1, nrows=31)
test = pd.read_excel(path, usecols="F:G", skiprows=1, nrows=3).rename(columns=lambda x: x.split('.')[0])

def count_occurrences(string, pattern="+-+"):
    return sum(1 for i in range(len(string) - len(pattern) + 1) if string[i:i + len(pattern)] == pattern)

grouped = input.groupby("Product")["Result"].agg(''.join).reset_index()
grouped['Number of repitation'] = grouped['Result'].apply(count_occurrences)
grouped.drop(columns="Result", inplace=True)

print(grouped.equals(test)) # True

File: files/100-199/135/CH-135.R
library(tidyverse)
library(readxl)

path = "files/CH-135 Identify the Pattern.xlsx"
input = read_excel(path, range = "B2:D32")
test  = read_excel(path, range = "F2:G5")

count_occurences = function(string, pattern = "+-+") {
  pattern_length = nchar(pattern)
  chars = unlist(strsplit(string, ""))
  matches = integer(0)
  for (i in seq_len(length(chars) - pattern_length + 1)) {
    segment = paste0(chars[i:(i + pattern_length - 1)], collapse = "")
    if (segment == pattern) {
      matches = c(matches, i)
    }
  }
  return(length(matches))
}

result = input %>%
  summarise(Result = str_c(Result, collapse = ""), .by = Product) %>%
  mutate(`Number of repitation` = map_int(Result, count_occurences)) %>%
  select(Product, `Number of repitation`) %>%

all.equal(result, test, check.attributes = FALSE)
#> [1] TRUE